apps/microtvm/zephyr/template_project/microtvm_api_server.py

```diff
@@ -152,6 +152,20 @@ def _get_flash_runner():
     return doc["flash-runner"]
 
 
+def _find_board_from_cmake_file(cmake_file: Union[str, pathlib.Path]) -> str:
+    """Find Zephyr board from generated CMakeLists.txt"""
+    zephyr_board = None
+    with open(cmake_file) as cmake_f:
+        for line in cmake_f:
+            if line.startswith("set(BOARD"):
+                zephyr_board = line.strip("\n").strip("\r").strip(")").split(" ")[1]
```

Comment:
I think it's call for a capture group here. For instance:
```
     zephyr_board = None
     with open(cmake_file) as cmake_f:
         for line in cmake_f:
-            if line.startswith("set(BOARD"):
-                zephyr_board = line.strip("\n").strip("\r").strip(")").split(" ")[1]
+            set_board = re.match("set\(BOARD (.*)\)", line)
+            if set_board:
+                zephyr_board = set_board.group(1)
                 break
 
     if not zephyr_board:
         raise RuntimeError(f"No Zephyr board set in the {cmake_file}.")
+
     return zephyr_board

```

Reply:
But since it's minor enhancement I won't block on this.

Reply:
thanks for the suggestion, I will fix this in a follow up PR